crates/oxc_linter/src/rules/unicorn/no_anonymous_default_export.rs

```diff
@@ -0,0 +1,155 @@
+            AstKind::ExportDefaultDeclaration(export_decl) => match &export_decl.declaration {
+                ExportDefaultDeclarationKind::ClassDeclaration(class_decl) => class_decl
+                    .id
+                    .as_ref()
+                    .map_or(Some((export_decl.span, ErrorNodeKind::Class)), |_| None),
+                ExportDefaultDeclarationKind::FunctionDeclaration(function_decl) => function_decl
+                    .id
+                    .as_ref()
+                    .map_or(Some((export_decl.span, ErrorNodeKind::Function)), |_| None),
+                ExportDefaultDeclarationKind::ArrowFunctionExpression(_) => {
+                    Some((export_decl.span, ErrorNodeKind::Function))
+                }
+                _ => None,
+            },
+            // CommonJS: module.exports
+            AstKind::AssignmentExpression(expr) if is_common_js_export(expr) => match &expr.right {
+                Expression::ClassExpression(class_expr) => {
+                    class_expr.id.as_ref().map_or(Some((expr.span, ErrorNodeKind::Class)), |_| None)
+                }
+                Expression::FunctionExpression(function_expr) => function_expr
+                    .id
+                    .as_ref()
+                    .map_or(Some((expr.span, ErrorNodeKind::Function)), |_| None),
+                Expression::ArrowFunctionExpression(_) => {
+                    Some((expr.span, ErrorNodeKind::Function))
+                }
+                _ => None,
+            },
+            _ => None,
+        };
+
+        if let Some((span, error_kind)) = problem_node {
+            ctx.diagnostic(NoAnonymousDefaultExportDiagnostic(span, error_kind.to_string()));
+        };
+    }
+}
+
+fn is_common_js_export(expr: &AssignmentExpression) -> bool {
+    if let AssignmentTarget::StaticMemberExpression(member_expr) = &expr.left {
+        if let Expression::Identifier(object_ident) = &member_expr.object {
+            if object_ident.name != "module" {
+                return false;
+            }
+        }
+
+        if member_expr.property.name != "exports" {
+            return false;
+        }
+    }
+
+    true
+}
+
+enum ErrorNodeKind {
+    Function,
+    Class,
+}
+
+impl fmt::Display for ErrorNodeKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let display_name = match self {
+            Self::Function => "function",
+            Self::Class => "class",
+        };
+        write!(f, "{display_name}")
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        r"export default class Foo {}",
+        r"export default function foo () {}",
+        r"const foo = () => {}; export default foo;",
+        r"module.exports = class Foo {};",
+        r"module.exports = function foo () {};",
+        r"const foo = () => {}; module.exports = foo;",
+        // TODO: need handle this situation?
+        // r"module['exports'] = function foo () {};",
```

Comment:
First off thanks for your contribution🥳.

I'm not sure about this one so I let others pitch in, but there is another case that fails,

```js
        export default (async function * () {})
        export default (class extends class {} {})
```

You also have to check for nested items within parenthesized expressions(and in theory can be nested infinitely).

Reply:
🤔 you mean that declaration of `export default` can be wrapped in any level parens?

Reply:
Yes exactly, I suggest you take a look at these cases from the original plugin(if you already haven't).
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v52.0.0/test/no-anonymous-default-export.mjs

Porting the tests from the original rule can help to prevent any unintentional difference between them.